matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/Matrix.kt

```diff
@@ -96,6 +96,11 @@ class Matrix private constructor(context: Context, matrixConfiguration: MatrixCo
 
     companion object {
 
+        /**
+         * Determines whether or not thread messages are enabled
+         */
+        var areThreadMessagesEnabled: Boolean = false
+
```

Comment:
I do not really like my solution here, I tried to find a better way using DI but didn't work as expected. What I already tried:
- Interface in **matrix-sdk**
- Implementation in **vector**
- Bind Implementation to Interface in vector module

Reply:
For global configuration of the SDK by the application, we use the class `MatrixConfiguration` if it helps.

Reply:
It does not because `MatrixConfiguration` is initiated too early on Application level before @Injections are available like userPreferences etc.  I think the best solution is to store it in sharedPref within the matrix SDK